storagemarket/types/contract_deal_proposal_types.go

```diff
@@ -7,11 +7,14 @@ import (
 	"github.com/ipfs/go-cid"
 )
 
+//go:generate cbor-gen-for --map-encoding ContractDealProposal ContractParamsVersion1
+
 type ContractDealProposal struct {
 	PieceCID     cid.Cid
 	PieceSize    abi.PaddedPieceSize
 	VerifiedDeal bool
 	Client       address.Address
+	Provider     string
```

Comment:
`Provider` should probably go into the `ContractParamsVersion1`, given that it is optional and given that we might want to change to a slice in the future, etc.